fakeredis/_basefakesocket.py

```diff
@@ -253,18 +255,20 @@ def _process_command(self, fields: List[bytes]):
         if not isinstance(result, NoResponse):
             self.put_response(result)
 
-    def _scan(self, keys, cursor, *args):
+    def _scan(self, keys, cursor, scan_type, *args):
```

Comment:
I don't think redis has a separate snapshot per scan-type.
Can you write a test for it?

Reply:
What kind of tests would you like? the scans are covered with tests already

Reply:
- Something similar to the one in your issue: rename/delete an item in the middle of a scan and see that the renamed/deleted item is still retrieved in the scan (a bit different than your original version).
 - Verify snapshot is renewed on cursor 0: A test starting a scan, deleting a key in the middle and then starting a new scan and seeing the deleted key is not part of the new scan.
 
- The tests are working in a way that checks real redis behavior and fakeredis behavior - It would be interesting to have a test with the scan within a scan to see the expected behavior from real redis. If it fails to match the expected behavior on fakeredis - you can mark it for now with the `@xfail` and we can address it later.

Reply:
By the way, I should have started with it: Thank you for the PR!

Reply:
Sure, will take me a couple of days